pytomorrowio/pytomorrowio.py

```diff
@@ -232,37 +174,34 @@ async def _forecast(
         start_time: Optional[datetime] = None,
         duration: Optional[timedelta] = None,
         **kwargs,
-    ) -> List[Dict[str, Any]]:
+    ) -> Dict[str, Any]:
         """Return forecast data from Tomorrow.io's API for a given time period."""
-        params = {
-            "fields": self.convert_fields_to_measurements(
-                process_v4_fields(fields, timestep)
-            ),
+        if timestep not in VALID_TIMESTEPS:
+            raise InvalidTimestep(f"{timestep} is not a valid 'timestep' parameter")
+        fields = process_v4_fields(fields, timestep)
+        if timestep > ONE_HOUR:
+            fields = self.convert_fields_to_measurements(fields)
+
+        params: Dict[str, Any] = {
+            "fields": fields,
             **kwargs,
         }
-        if timestep == timedelta(days=1):
-            params["timestep"] = [TIMESTEP_DAILY]
-        elif timestep == timedelta(hours=1):
-            params["timestep"] = [TIMESTEP_HOURLY]
-        elif timestep in (
-            timedelta(minutes=30),
-            timedelta(minutes=15),
-            timedelta(minutes=5),
-            timedelta(minutes=1),
-        ):
-            params["timestep"] = [f"{int(timestep.total_seconds()/60)}m"]
+        if timestep == ONE_DAY:
+            params["timesteps"] = [TIMESTEP_DAILY]
```

Comment:
New unit test found that hourly results were always being returned because `timesteps` is the param name--not `timestep`.

Reply:
nice catch!